azurerm/internal/services/powerbidedicated/resource_arm_powerbi_dedicated_capacity.go

```diff
@@ -0,0 +1,236 @@
+					"A2",
+					"A3",
+					"A4",
+					"A5",
+					"A6",
+				}, false),
+			},
+
+			"administrators": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: ValidateCapacityAdministratorName,
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmPowerBIDedicatedCapacityCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).PowerBIDedicated.CapacityClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() {
+		existing, err := client.GetDetails(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_powerbi_dedicated_capacity", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	administrators := d.Get("administrators").(*schema.Set).List()
+	skuName := d.Get("sku_name").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := powerbidedicated.DedicatedCapacity{
+		Location: utils.String(location),
+		DedicatedCapacityProperties: &powerbidedicated.DedicatedCapacityProperties{
+			Administration: &powerbidedicated.DedicatedCapacityAdministrators{
+				Members: utils.ExpandStringSlice(administrators),
+			},
+		},
+		Sku: &powerbidedicated.ResourceSku{
+			Name: utils.String(skuName),
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.Create(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read Capacity %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmPowerBIDedicatedCapacityRead(d, meta)
+}
+
+func resourceArmPowerBIDedicatedCapacityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).PowerBIDedicated.CapacityClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["capacities"]
```

Comment:
Could we write a parse function for this resource as can be seen in #5779

Reply:
Updated